c7n/resources/iam.py

```diff
@@ -792,8 +792,17 @@ def get_evaluations(self, client, arn, r, actions):
             if boundary_policy:
                 params['PermissionsBoundaryPolicyInputList'] = [boundary_policy]
 
-        evaluations = self.manager.retry(
-            client.simulate_principal_policy, **params).get('EvaluationResults', ())
+        try:
+            evaluations = self.manager.retry(
```

Comment:
we should be able to ignore the below exception too:

```python
ignore_err_codes=('NoSuchEntityException',))
```

Reply:
Good point, and we do use that pattern lots in the iam resource. Though that way we lose the ability to add the conditional warning below about possible invalid weak references.

Then again, maybe that sort of warning isn't particularly useful? Is it better to let a dedicated `invalid` filter catch broken weak references and silently ignore them here?

Reply:
warning logs at scale aren't very readable, mostly due to the fact that users won't typically take a look at successful policy executions, they usually care about the unsuccessful ones ;). I like your idea of a dedicated invalid filter as well

Reply:
Makes sense, fair point - removed the explicit try/except and warning in favor of using `ignore_err_codes`. The `invalid` filter came up in previous discussions and seems in line with how we handle [broken autoscaling groups](https://github.com/cloud-custodian/cloud-custodian/blob/42d011df2fbda2141e2eb84197adaed8d02422a8/c7n/resources/asg.py#L391-L417).